PsfLauncher/main.cpp

```diff
@@ -382,9 +441,18 @@ ErrorInformation StartProcess(ExecutionInformation execInfo, int cmdShow, bool r
         (LPSTARTUPINFO)&startupInfoEx,
         &processInfo))
     {
-        DWORD waitResult = ::WaitForSingleObject(processInfo.hProcess, INFINITE);
-
-        if (waitResult != WAIT_OBJECT_0)
+        DWORD waitResult = ::WaitForSingleObject(processInfo.hProcess, timeout);
+
+		if (waitResult == WAIT_TIMEOUT)
+		{
+			//Terminating the process is not considered an exception.
+			if (!TerminateProcess(processInfo.hProcess, ERROR_PROCESS_ABORTED))
+			{
+				return { L"Error stopping the process.", ::GetLastError() };
+			}
+		}
+
+        if (waitResult == WAIT_FAILED || waitResult == WAIT_ABANDONED)

```

Comment:
Does WAIT_ABANDONED set the lasterror?

Reply:
No it does not.

Reply:
Then this logic is wrong. You should move WAIT_ABANDONED out and return a different error.